### PR TITLE
Custom fee labels

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

Follow up on https://github.com/cowprotocol/cowswap/pull/4330

This PR allows users of the widget to customize the tooltip and add any content they want. 

Screenshot at Apr 29 19-01-22

Screenshot at Apr 29 19-02-54

## Config

The configuration allows you to provide a custom fee label and the markdown for the tooltip

```ts
{
content: {
        // TODO: Don't merge with this. This comment should be reverted.
        feeLabel: "Anxo's cut",
        feeTooltipMarkdown: `\
Anxo has been working hard lately, and need to pay his bills.
Anxo is a good guy, so you should pay him.
This fee won't contribute to anything useful, but it will make Anxo happy.
## Why
Think about it
## Read more
[don't read more](google.com)
`,
      },
}
```

## Fun fact
Copilot helped writing the tooltip message, so I think IA wants me to be payed more

## Test

1. Go to configurator https://widget-configurator-git-custom-fee-labels-cowswap.vercel.app/
2. Check the fee label
3. Check the fee tooltip

## Additional
DONT'T MERGE: Needs to revert a commit that modifies the configurator label before merging

### CODE REVIEW

**Review by Dioliode — APPROVED** [review]
DioX

**Review by alfetopito — COMMENTED** [review]
It's not possible to hover over the tooltip to click on 'read more'

image

File: apps/cowswap-frontend/src/modules/swap/containers/Row/RowFee/index.tsx [hunk 1]
```diff
@@ -157,16 +161,18 @@ export function RowPartnerFee({ partnerFee, feeAmount, feeInFiat }: PartnerRowPa
     const { bps } = partnerFee || { bps: 0 }
     const isFree = bps === 0
 
+    const markdownContent = tooltipMarkdown ? <ReactMarkdown components={markdownComponents as Components}>{tooltipMarkdown}</ReactMarkdown> : undefined
+
     return {
-      label: isFree ? 'Fee' : 'Total fee',
+      label: label ? label : isFree ? 'Fee' : 'Total fee',
```
[the remark this answers is not in the document]
Comment (anxolin) [reply apps/cowswap-frontend/src/modules/swap/containers/Row/RowFee/index.tsx:167]: My thinking is, the user of the widget will provide both the BPS and the label. So if they want this behavior they can still have it, but providing the word 'fee' or `undefined` when the BPS=0

Its not true the other way around, if I let the `isFree` to take precedence, they can't control the label anymore when BPS is zero

**Comment by anxolin** [comment]
> It's not possible to hover over the tooltip to click on 'read more'

@alfetopito , it should be, this is why I implemented before this PR https://github.com/cowprotocol/cowswap/pull/4330

File: apps/cowswap-frontend/src/modules/swap/containers/Row/RowFee/index.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (alfetopito) [reply apps/cowswap-frontend/src/modules/swap/containers/Row/RowFee/index.tsx:167]: Fair enough.
Then they should be aware of the precedence for the "Free" to be displayed.

**Comment by alfetopito** [comment]
> > It's not possible to hover over the tooltip to click on 'read more'
> 
> @alfetopito , it should be, this is why I implemented before this PR #4330

Right! I reviewed that one afterwards.
It was not obvious to me I had to click on the icon, that's why I thought it wasn't working.

**Review by alfetopito — APPROVED** [review]
Approved with the caveat noted in the other PR that's not clear how to make the tooltips sticky.

**Review by elena-zh — APPROVED** [review]
Hey @anxolin , great job!!
However, same as Leandro, for me it is not so obvious that I need to click on the tooltip icon in order to catch the link inside.

**Pull request merged**